components/calendar/Header.tsx

```diff
@@ -128,37 +142,65 @@ export default class Header extends React.Component<HeaderProps, any> {
     }
   };
 
-  onTypeChange = (e: RadioChangeEvent) => {
+  onTypeChange = (type: string) => {
     const onTypeChange = this.props.onTypeChange;
     if (onTypeChange) {
-      onTypeChange(e.target.value);
+      onTypeChange(type);
     }
   };
 
   getCalenderHeaderNode = (node: HTMLDivElement) => {
     this.calenderHeaderNode = node;
   };
 
-  renderHeader = ({ getPrefixCls }: ConfigConsumerProps) => {
-    const { prefixCls: customizePrefixCls, type, value, locale, fullscreen } = this.props;
+  getMonthYearSelections = (getPrefixCls: ConfigConsumerProps['getPrefixCls']) => {
+    const { prefixCls: customizePrefixCls, type, value } = this.props;
+
     const prefixCls = getPrefixCls('fullcalendar', customizePrefixCls);
-    const yearSelect = this.getYearSelectElement(prefixCls, value.year());
-    const monthSelect =
+    const yearReactNode = this.getYearSelectElement(prefixCls, value.year());
+    const monthReactNode =
       type === 'month'
         ? this.getMonthSelectElement(prefixCls, value.month(), this.getMonthsLocale(value))
         : null;
+    return {
+      yearReactNode,
+      monthReactNode,
+    };
+  };
+
+  getTypeSwitch = () => {
+    const { locale, type, fullscreen } = this.props;
     const size = fullscreen ? 'default' : 'small';
-    const typeSwitch = (
-      <Group onChange={this.onTypeChange} value={type} size={size}>
+    return (
+      <Group onChange={e => this.onTypeChange(e.target.value)} value={type} size={size}>
         <Button value="month">{locale.month}</Button>
         <Button value="year">{locale.year}</Button>
       </Group>
     );
+  };
+
+  headerRenderCustom = (): React.ReactNode => {
+    const { headerRender, type, onValueChange, value } = this.props;
+    return (
+      <div ref={this.getCalenderHeaderNode}>
```

Comment:
No need to set ref here since customize code not reach the `calenderHeaderNode`.